src/core/qgsfileutils.h

```diff
@@ -118,6 +119,38 @@ class CORE_EXPORT QgsFileUtils
      */
     static QStringList findFile( const QString &file, const QString &basepath = QString(), int maxClimbs = 4, int searchCeiling = 4, const QString &currentDir = QString() );
 
+    /**
+     * Drive types
+     * \since QGIS 3.20
+     */
+    enum DriveType
+    {
+      Unknown, //!< Unknown type
+      Invalid, //!< Invalid path
+      Removable, //!< Removable drive
+      Fixed, //!< Fixed drive
+      Remote, //!< Remote drive
+      CdRom, //!< CD-ROM
+      RamDisk, //!< RAM disk
+    };
+
+    /**
+     * Returns the drive type for the given \a path.
+     *
+     * \throws QgsNotSupportedException if determining the drive type is not supported on the current platform.
+     *
+     * \since QGIS 3.20
+     */
+    static DriveType driveType( const QString &path ) SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns TRUE if the specified \a path is known to reside on a slow device, e.g. a remote
```

Comment:
```suggestion
     * Returns TRUE if the specified \a path is assumed to reside on a slow device, e.g. a remote
```

Reply:
I think "known" is more accurate here -- e.g. for any non-windows platform we always return false, because we don't know either way.

Reply:
I am hesitating to call results of heuristics "known" fter all it depends on factors like network speed etc.

Reply:
Changed